alg/sha256.c

```diff
@@ -80,7 +84,7 @@ static const uint32_t initial_state[8] = {
 	0x510E527F, 0x9B05688C, 0x1F83D9AB, 0x5BE0CD19
 };
 
-#if defined(CPUSUPPORT_X86_SHANI) && defined(CPUSUPPORT_X86_SSSE3)
+#ifdef SELFTEST
 /*
  * Test whether software and SHANI transform code produce the same results.
```

Comment:
Did you want to rename "shanitest" to "hwtest" or something similar while you're "genericizing" this?

Reply:
Nah, not really.  I could do it if you think it's worth it.

Reply:
Ok, just checking.  We can always change it when we add non-x86 bits.

Reply:
ah, right.  I only glanced at `crc32.c`, for which `crctest()` would still be a valid name.  But you're right that `shanitest()` wouldn't be appropriate for ARM SHA code.

Ok, I'll change them all to hwtest.